packages/web-runtime/src/components/Topbar/Notifications.vue

```diff
@@ -1,117 +1,337 @@
+import { formatDateFromISO, formatRelativeDateFromISO, useClientService, useStore } from 'web-pkg'
+import { useGettext } from 'vue3-gettext'
+import { useTask } from 'vue-concurrency'
+
+const POLLING_INTERVAL = 30000
 
 export default {
   components: {
     NotificationBell
   },
-  computed: {
-    ...mapGetters(['activeNotifications', 'configuration'])
-  },
-  methods: {
-    ...mapActions(['deleteNotification', 'showMessage']),
+  setup() {
+    const store = useStore()
+    const { owncloudSdk } = useClientService()
+    const { current: currentLanguage } = useGettext()
+
+    const notifications: Ref<Notification[]> = ref([])
+    const loading = ref(false)
+    const notificationsInterval = ref()
+    const dropdownOpened = ref(false)
+
+    const formatDate = (date) => {
+      return formatDateFromISO(date, currentLanguage)
+    }
+    const formatDateRelative = (date) => {
+      return formatRelativeDateFromISO(date, currentLanguage)
+    }
+
+    const messageParameters = [
+      { name: 'user', labelAttribute: 'displayname' },
+      { name: 'resource', labelAttribute: 'name' },
+      { name: 'space', labelAttribute: 'name' }
+    ]
+    const getMessage = ({ message, messageRich, messageRichParameters }: Notification) => {
+      if (messageRich) {
+        let interpolatedMessage = messageRich
+        for (const param of messageParameters) {
+          if (interpolatedMessage.includes(`{${param.name}}`)) {
+            const label = messageRichParameters[param.name][param.labelAttribute]
+            interpolatedMessage = interpolatedMessage.replace(
+              `{${param.name}}`,
+              `<strong>${label}</strong>`
+            )
+          }
+        }
+        return interpolatedMessage
+      }
+      return message
+    }
+    const getLink = ({ messageRichParameters, object_type }: Notification) => {
+      if (!messageRichParameters) {
+        return null
+      }
+      if (object_type === 'share' && messageRichParameters?.resource?.id) {
+        return {
+          name: 'files-shares-with-me',
+          query: {
+            scrollTo: messageRichParameters.resource.name
+          }
+        }
+      }
+      if (object_type === 'storagespace' && messageRichParameters?.space?.id) {
+        const space = store.getters['runtime/spaces/spaces'].find(
+          (s) => s.fileId === messageRichParameters?.space?.id.split('!')[0]
+        )
+        if (space) {
+          return { name: 'files-spaces-generic', params: { driveAliasAndItem: space.driveAlias } }
+        }
+      }
+      return null
+    }
 
-    executeRequest(app, link, type, notificationId) {
-      this.$client.requests
-        .ocs({
+    const setAdditionalData = () => {
+      loading.value = true
+      for (const notification of unref(notifications)) {
+        notification.computedMessage = getMessage(notification)
+        notification.computedLink = getLink(notification)
+      }
+      loading.value = false
+    }
```

Comment:
I decided to separate the additional data from the `fetchNotificationsTask` and only load it when the notifications drop menu is open. The reason behind this is because I don't now how expensive it will get with a bunch of notifications. Web polls notifications actively every 30 seconds, there is no need to load additional data every time.

I don't see any disadvantages with that approach, but I also don't now if it's necessary. Feedback welcome!